### PR TITLE
Add blsct::{PublicKey,DoublePublicKey,PrivateKey} classes

### PR DESCRIPTION
**Pull request by aguycalled**
Solves #28 

Adds the following classes and corresponding tests:

- `blsct::PublicKey`: A G1Point representing a BLS Public Key
- `blsct::DoublePublicKey`: BLSCT Double Public Keys (representing 1 View BLS Public Key + 1 Spend BLS Public Key)
- `blsct::PrivateKey`: A Scalar representing a BLS Private Key

Pending to do in future tasks: #55 and #63

### CODE REVIEW

File: src/blsct/keys.cpp [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+PublicKey PrivateKey::GetPublicKey() const
+{
+    return PublicKey(GetG1Point());
+}
+
+Scalar PrivateKey::GetScalar() const
+{
+    return Scalar(std::vector<unsigned char>(k.begin(), k.end()));
+    ;
```
Comment (gogoex) [inline src/blsct/keys.cpp:151]: This can be deleted

File: src/test/blsct/keys_tests.cpp [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    blsct::PrivateKey privateKeyFromAddition(privateKeyFromScalar.GetScalar() +
+                                             privateKeyFromVector.GetScalar());
+    blsct::PublicKey publicKeyFromAddition(privateKeyFromScalar.GetG1Point() +
+                                           privateKeyFromVector.GetG1Point());
+    BOOST_CHECK(privateKeyFromAddition.GetPublicKey() == publicKeyFromAddition);
+
+    std::vector<blsct::PublicKey> vecKeys = {privateKeyFromScalar.GetPublicKey(), privateKeyFromVector.GetPublicKey()};
+    BOOST_CHECK(blsct::PublicKey::Aggregate(vecKeys) == publicKeyFromAddition);
+}
+
```
Comment (gogoex) [inline src/test/blsct/keys_tests.cpp:123]: It would be nicer if the test checks if the serialization order of DoublePublicKey is vk followed by sk too.

**Comment by mxaddict** [comment]
Built and ran testsuite on Ubuntu 20.04 using WSL2.

**Comment by mxaddict** [comment]
Linter failed with:
```
The test suite in file src/test/foo_tests.cpp should be named
"foo_tests". Please make sure the following test suites follow
that convention:

src/test/blsct/keys_tests.cpp:BOOST_FIXTURE_TEST_SUITE(blsct_keys_tests, MclTestingSetup)

Success: no issues found in 243 source files
^---- failure generated from lint-tests.py
```

**Comment by mxaddict** [comment]
> Linter failed with:
> 
> ```
> The test suite in file src/test/foo_tests.cpp should be named
> "foo_tests". Please make sure the following test suites follow
> that convention:
> 
> src/test/blsct/keys_tests.cpp:BOOST_FIXTURE_TEST_SUITE(blsct_keys_tests, MclTestingSetup)
> 
> Success: no issues found in 243 source files
> ^---- failure generated from lint-tests.py
> ```

I think it just needs to rename the test suite name

**Review by mxaddict — DISMISSED** [review]

File: src/test/blsct/keys_tests.cpp (hunk 2 above)
Comment (aguycalled) [reply src/test/blsct/keys_tests.cpp:129]: addressed in [5066426](https://github.com/navcoin/navcoin/pull/62/commits/506642627ae82066d7240b4834bcee208df2a51e)

**Review by mxaddict — DISMISSED** [review]

**Comment by sinetek** [comment]
Looks good to me. There's some mixed usage of  vector<uint8_t>  // vector<unsigned char>, and the use of .begin() .end() everywhere is a bit barbaric but ok.

**Comment by aguycalled** [comment]
> Looks good to me. There's some mixed usage of vector<uint8_t> // vector< unsigned char >, and the use of .begin() .end() everywhere is a bit barbaric but ok.

Substituted `uint8_t` with `unsigned char` in [6d596bf](https://github.com/navcoin/navcoin/pull/62/commits/6d596bf62ca9098bb6990e5d1e54a37d4401b966)

What would you suggest as an alternative for the `.begin()` and `.end()` uses?

**Comment by sinetek** [comment]
> > Looks good to me. There's some mixed usage of vector<uint8_t> // vector< unsigned char >, and the use of .begin() .end() everywhere is a bit barbaric but ok.
> 
> Substituted `uint8_t` with `unsigned char` in [6d596bf](https://github.com/navcoin/navcoin/pull/62/commits/6d596bf62ca9098bb6990e5d1e54a37d4401b966)
> 
> What would you suggest as an alternative for the `.begin()` and `.end()` uses?

These are minor nits, I'm not a fan of the old iterators, C++20 syntax is an improvement, or perhaps Boost.Range if there's a lot of range operations.

**Review by gogoex — APPROVED** [review]

**Comment by mxaddict** [comment]
> 

I'm interested in the C++20 syntax changes as well, I've not used them yet.

How would you for example write the vector constructor copy in new syntax?

**Pull request merged**